### PR TITLE
Test for EXT_texture_norm16 extension

### PR DESCRIPTION
**Pull request by shrekshao**
Add test for EXT_texture_norm16 extension.
Refactor checkCanvasRect a bit to expose `opt_readBackFormat` to assign formats other than `gl.RGBA`

### CODE REVIEW

File: sdk/tests/conformance2/extensions/ext-texture-norm16.html [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+<!-- <script src="../../js/tests/compressed-texture-utils.js"></script> -->
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description("This test verifies the functionality of the EXT_texture_compression_rgtc extension, if it is available.");
```
Comment (jdashg) [inline sdk/tests/conformance2/extensions/ext-texture-norm16.html]: s/EXT_texture_compression_rgtc/EXT_texture_norm16/

File: sdk/tests/conformance2/extensions/00_test_list.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ext-color-buffer-float.html
 ext-disjoint-timer-query-webgl2.html
 --min-version 2.0.1 ext-texture-filter-anisotropic.html
+--min-version 2.0.1 ext-texture-norm16.html
```
Comment (shrekshao) [inline sdk/tests/conformance2/extensions/00_test_list.txt]: Expose to WebGL 1

Comment (kenrussell) [reply sdk/tests/conformance2/extensions/00_test_list.txt]: Per discussion in WebGL WG today, let's leave this as a WebGL 2.0-specific extension for the initial iteration.

File: sdk/tests/conformance2/extensions/ext-texture-norm16.html (hunk 1 above)
Comment (shrekshao) [reply sdk/tests/conformance2/extensions/ext-texture-norm16.html]: Done. Thanks for the Review!

**Pull request merged**